test/fs-cache.js

```diff
@@ -0,0 +1,142 @@
+const assert = require('assert');
+const path = require('path');
+const rimraf = require('rimraf');
+const fs = require('../src/utils/fs');
+const promisify = require('../src/utils/promisify');
+const {sleep} = require('./utils');
+const ncp = promisify(require('ncp'));
+const FSCache = require('../src/FSCache');
+
+const cachePath = path.join(__dirname, '.cache');
+const inputPath = path.join(__dirname, '/input');
+
+const getMTime = async file => {
+  const stat = await fs.stat(file);
+  const mtime = stat.mtime.getTime();
+  return mtime;
+};
+
+describe('prettyError', () => {
+  beforeEach(() => {
+    rimraf.sync(cachePath);
+    rimraf.sync(inputPath);
+  });
+
+  it('should create directory on ensureDirExists', async () => {
+    let exists = await fs.exists(cachePath);
+    assert(!exists);
+
+    const cache = new FSCache({cacheDir: cachePath});
+    await cache.ensureDirExists();
+
+    exists = await fs.exists(cachePath);
+    assert(exists);
+  });
+
+  it('should cache resources', async () => {
+    const cache = new FSCache({cacheDir: cachePath});
+    await cache.write(__filename, {a: 'test', b: 1, dependencies: []});
+
+    let cached = await cache.read(__filename);
+    assert.equal(cached.a, 'test');
+    assert.equal(cached.b, 1);
+  });
+
+  it('should return null for invalidated resources', async () => {
+    const cache = new FSCache({cacheDir: cachePath});
+    cache.invalidate(__filename);
+
+    let cached = await cache.read(__filename);
+    assert.equal(cached, null);
+  });
+
+  it('should remove file on delete', async () => {
```

Comment:
`cache.delete` was not working due to `fs.unlink` missing from the `fs` utility. The implementation catches the error and fails silently. Does not appear to be used today, but is now working correctly. 

Exposed by tests, fixed.